homeassistant/components/media_player/yamaha.py

```diff
@@ -48,9 +60,26 @@ def update(self):
         else:
             self._pwstate = STATE_OFF
         self._muted = self._receiver.mute
-        self._volume = (self._receiver.volume/100) + 1
-        self._current_source = self._receiver.input
-        self._source_list = list(self._receiver.inputs().keys())
+        self._volume = (self._receiver.volume / 100) + 1
+
+        if self.source_list is None:
+            self.build_source_list()
+
+        current_source = self._receiver.input
+        if current_source in self._source_names:
```

Comment:
Dictionaries in Python support a method that does exactly this and the next 3 lines:
```python
self._current_source = self._source_names.get(current_source, current_source)
```

The second argument to `get` is the value to return if the key was not found.